### PR TITLE
Clean-up in 3/1 - 3/4

### PR DESCRIPTION
**Pull request by bast**
- [x] book in sync
- [x] ready to go

### CODE REVIEW

**Comment by bast** [comment]
This is ready for review. Perhaps good if we don't accumulate a gigantic PR. I will submit the other recipes as separate PR (I expect fewer changes there).

File: chapter-03/recipe-01/example/CMakeLists.txt [hunk 1]
```diff
@@ -16,5 +16,10 @@ execute_process(
   ERROR_QUIET
   OUTPUT_STRIP_TRAILING_WHITESPACE
   )
+
 message(STATUS "RESULT_VARIABLE is: ${_status}")
 message(STATUS "OUTPUT_VARIABLE is: ${_hello_world}")
+
+# compare the "manual" messages with the following handy helper
+include(CMakePrintHelpers)
```
Comment (robertodr) [inline chapter-03/recipe-01/example/CMakeLists.txt]: Glad you managed to squeeze this one in!

File: chapter-03/recipe-03/cxx-example/CMakeLists.txt [hunk 2]
```diff
@@ -39,31 +41,42 @@ find_package_handle_standard_args(NumPy
...
-else()
-  target_sources(pure-embedding
-    PRIVATE
-      Py3-pure-embedding.cpp
-    )
-endif()
+
+target_sources(pure-embedding
+  PRIVATE
+    Py${PYTHON_VERSION_MAJOR}-pure-embedding.cpp
```
Comment (robertodr) [inline chapter-03/recipe-03/cxx-example/CMakeLists.txt]: Indeed... How could I not see this? :man_facepalming:

File: chapter-03/recipe-04/cxx-example/linear-algebra.cpp [hunk 3]
```diff
@@ -4,11 +4,8 @@
 #include "CxxBLAS.hpp"
 #include "CxxLAPACK.hpp"
 
-#include <chrono>
-#include <cstdlib>
 #include <iostream>
 #include <random>
-#include <vector>
```
Comment (robertodr) [inline chapter-03/recipe-04/cxx-example/linear-algebra.cpp]: Why remove the `#include <vector>`? It's still used in the code.

Comment (bast) [reply chapter-03/recipe-04/cxx-example/linear-algebra.cpp]: OK I will check, I was possibly too quick.

**Comment by bast** [comment]
I will fix the vector and check clang-format and send a patch.

**Comment by robertodr** [comment]
Both travis and appveyor are happy with this. So I suggest we put those two changes in with the next PR.

**Comment by bast** [comment]
OK - you can merge - I will add these changes to the second set of recipes.

**Comment by bast** [comment]
Thanks for catching the vector - not sure why I removed it. I blame it on lack of caffeine. It is part of upcoming PR.

**Comment by robertodr** [comment]
:coffee:

**Pull request merged**